chainerio/__init__.py

```diff
@@ -68,7 +122,53 @@ def open(file_path: str, mode: str = 'rb',
         errors, newline, closefd, opener)
 
 
-def set_root(uri_or_handler: str) -> None:
+def set_root(uri_or_handler: Union[str, Type['IO']]) -> None:
+    """set the current context to uri_or_handler
+
+    The context here refers to the ``default handler`` and the ``root path``.
+    The ``default handler`` points to a filesystem or container which
+    chainerio uses when called without a full uri. Overrides by the full uri.
+
+    # Example
+
+    chainer.set_root("posix")
+    # open a file on posix filesystem with path "some/file"
+    chainerio.open("some/file")
+    # override with a full uri
+    chainerio.open("hdfs:///some/file/on/hdfs")
+
+    The ``root path`` refers to a directory or a file that chainerio works on.
+    Similar to the ``CWD``. Note that the ``root path`` can be set to a file
+    for when it points to a container. The ``root path`` will only be set
+    when the ``uri_or_handler`` points to directory.
+    Otherwise, it will be set to an empty string,
+    which represents the default working directory as follow:
+
+    +---------+---------------------------+
+    |         |     Default Root Path     |
+    +=========+===========================+
+    | POSIX   |    Current directory      |
+    +---------+---------------------------+
+    | HDFS    |     /user/USERNAME        |
```

Comment:
Maybe it would be nice if the USERNAME is clarified e.g.linking to Hadoop doc https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html

Reply:
Would be nice if it's quoted

Reply:
Addressed in e7c0806